apache2/msc_multipart.c

```diff
@@ -130,7 +134,41 @@ static int multipart_parse_content_disposition(modsec_rec *msr, char *c_d_value)
          */
 
         char quote = '\0';
-        if ((*p == '"') || (*p == '\'')) {
+        if (name == "filename*") {
+            /* filename*=charset'[optional-language]'filename */
+            /* Read beyond the charset and the optional language*/
+            const char* start_of_charset = p;
+            while ((*p != '\0') && (isalnum(*p) || (strchr(mime_charset_special, *p)))) {
+                p++;
+            }
+            if ((*p != '\'') || (p == start_of_charset)) {
+                return -16; // Must be at least one legit char before ' for start of language
+            }
+            p++;
+            while ((*p != '\0') && (*p != '\'')) {
```

Comment:
Languages can only contain alphanum & '-'
`while (isalnum(*p) || *p == '-') p++;`

Reply:
You're right, and here - as you suggested - we can control this.

The question is: is it allowed to send the request without charset? I mean:

```
Content-Disposition: form-data; name="post"; filename*=resume.pdf
```

I can't find any relevant information about that.

Reply:
So, do you think we do not need to check the charset and we should allow anything? Or restrict the charset content only to alnum chars (+ `-`)?

Reply:
IMO, the charset should be restricted to the ABNF from the RFC.